src/components/IdleStatusMonitor.js

```diff
@@ -33,19 +35,25 @@ const getIdleData = ({ currentTime, lastRecordedActivity, timeout, countdownStar
 
 const IdleStatusMonitor = ({
   timeout = Utils.durationToMillis({ minutes: 15 }),
-  countdownStart = Utils.durationToMillis({ minutes: 3 }), emailDomain = ''
+  countdownStart = Utils.durationToMillis({ minutes: 3 })
 }) => {
-  const { isSignedIn, user: { id, email } } = Utils.useStore(authStore)
-  // Placeholder code until the clinical user information is available in the user's profile.
-  // This will likely be a boolean property in the profile
-  const isClinicalDomain = email && emailDomain && email.endsWith(`@${emailDomain}`)
+  const signal = Utils.useCancellation()
+  const [isClinicalUser, setIsClinicalUser] = useState()
+  const { isSignedIn, registrationStatus, user: { id } } = Utils.useStore(authStore)
   const { [id]: lastRecordedActivity } = Utils.useStore(lastActiveTimeStore)
   const { timedOut } = getIdleData({ currentTime: Date.now(), lastRecordedActivity, timeout, countdownStart })
 
+  useEffect(() => {
+    const setClinicalStatus = withErrorReporting('Error loading group list', async () => {
+      setIsClinicalUser(_.some({ groupName: 'session_timeout' }, await Ajax(signal).Groups.list()))
+    })
+    isSignedIn && registrationStatus === 'registered' ? setClinicalStatus() : setIsClinicalUser()
```

Comment:
TOL: I think this logic will work, but we should get in the habit of being rigorous about what we're checking for in effects, because it's easy to get tripped up. For example, suppose that `signal` changed value (it won't, but just to illustrate), the effect would re-run, this check would succeed and we'd re-request the groups even though the person hadn't logged out.

What we actually want to detect is when `registrationStatus` transitions from `false` to `true`. To do that explicitly, we'd need to examine both the previous and current values. See the cascade of calls that happen in `auth.js` for some examples of that. Actually, thinking about it, I wonder if it would make sense for this call to happen in that file.

Reply:
+1 to moving the check to `auth.js`. It means we have to make `clinicalStatus` global, but that could be useful in other places like the Contact Us widget. We'd also have to remember to clear the global state on sign-out, which is also centralized in `auth.js`.

Reply:
I moved this to auth.js - it does look & feel much cleaner - thanks!